src/main/java/com/cloudbees/jenkins/plugins/bitbucket/api/credentials/BitbucketOAuthAuthenticator.java

```diff
@@ -11,20 +14,29 @@
 public class BitbucketOAuthAuthenticator extends BitbucketAuthenticator {
 
     private Token token;
+    private static final Logger LOGGER = Logger.getLogger(BitbucketOAuthAuthenticator.class.getName());
 
     /**
      * Constructor.
      *
      * @param credentials the key/pass that will be used
      */
-    public BitbucketOAuthAuthenticator(StandardUsernamePasswordCredentials credentials) {
+    public BitbucketOAuthAuthenticator(BitbucketOAuthCredentials credentials) {
         super(credentials);
 
-        OAuthConfig config = new OAuthConfig(credentials.getUsername(), Secret.toString(credentials.getPassword()));
+        OAuthConfig config;
+        try {
+            config = new OAuthConfig(credentials.getKey(), Secret.toString(credentials.getSecret()));
+
+            BitbucketOAuthService OAuthService = (BitbucketOAuthService) new BitbucketOAuth().createService(config);
+
+            token = OAuthService.getAccessToken(OAuthConstants.EMPTY_TOKEN, null);
+        } catch (IOException | InterruptedException e) {
+            // TODO Auto-generated catch block
+            LOGGER.log(Level.WARNING, "Could not retrieve credentials: { " + e.getMessage() + "}");
+        }
```

Comment:
`BitbucketOAuthCredentials.getSecret()` and `BitbucketOAuthCredentials.getKey()` are declared as possibly throwing these exceptions.  Are those methods the only reason, this catch is needed?  

I ask because it looks like the implementation of those methods never throw those exceptions.

Reply:
Previous implementation had overlaps with other credential types. ie AWS credentials.

Reply:
Actually that `BitbucketOAuthCredentials.getKey()` in declared as such in on error. But for `BitbucketOAuthCredentials.getSecret()` it's recommended to do so in the guide in order to allow the credential to be used by any `CredentialsProvider` storing credentials externally and using the `java.lang.reflect.Proxy` mechanism